controllers/dogu_requeue_handler_test.go

```diff
@@ -1,230 +1,237 @@
 package controllers
 
 import (
-	"context"
-	"errors"
+	"fmt"
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/mock"
-	"github.com/stretchr/testify/require"
-	v1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	fake2 "k8s.io/client-go/kubernetes/fake"
-	"k8s.io/client-go/rest"
-	ctrl "sigs.k8s.io/controller-runtime"
-
 	doguv2 "github.com/cloudogu/k8s-dogu-lib/v2/api/v2"
+	"github.com/cloudogu/k8s-dogu-lib/v2/client"
+	"github.com/cloudogu/k8s-dogu-operator/v3/controllers/config"
+	"github.com/stretchr/testify/assert"
+	v2 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
+	controllerruntime "sigs.k8s.io/controller-runtime"
 )
 
-type myRequeueableError struct{}
-
-func (mre myRequeueableError) Requeue() bool {
-	return true
-}
-
-func (mre myRequeueableError) Error() string {
-	return "my requeueable error"
-}
-
-func TestDoguRequeueHandler_Handle(t *testing.T) {
-	namespace := "namespace"
-
-	t.Run("handle no error at all", func(t *testing.T) {
-		// given
-		doguResource := &doguv2.Dogu{
-			ObjectMeta: metav1.ObjectMeta{Name: "myName", Labels: map[string]string{"test": "false"}},
-			Status:     doguv2.DoguStatus{},
-		}
-		fakeNonCacheClient := fake2.NewSimpleClientset()
-		eventRecorder := &mockEventRecorder{}
-		doguInterfaceMock := newMockDoguInterface(t)
-
-		handler := doguRequeueHandler{
-			doguInterface:  doguInterfaceMock,
-			nonCacheClient: fakeNonCacheClient,
-			namespace:      namespace,
-			recorder:       eventRecorder,
-		}
-
-		onRequeue := func(doguResource *doguv2.Dogu) error { doguResource.Labels["test"] = "true"; return nil }
-
-		// when
-		result, err := handler.Handle(context.Background(), "my context", doguResource, nil, onRequeue)
-
-		// then
-		require.NoError(t, err)
-
-		assert.Equal(t, result.RequeueAfter, time.Duration(0))
-
-		assert.Equal(t, "false", doguResource.Labels["test"])
-		mock.AssertExpectationsForObjects(t, eventRecorder)
-	})
-
-	t.Run("handle non reportable error", func(t *testing.T) {
+func TestNewDoguRequeueHandler(t *testing.T) {
+	t.Run("should fail to create DoguRequeueHandler", func(t *testing.T) {
```

Comment:
This test fails if a kube config exists on the hosts

Reply:
I think @nroeske fixed this by using the dependency injection.